docs/docs/core/database/relations/reordering.mdx

```diff
@@ -0,0 +1,116 @@
+tags:
+  - database
+  - relations
+  - reordering
+---
+
+Strapi allows to reorder a relation list.
+
+<img src="/img/database/reordering.png" alt="An example of reordering in the CM" />
+
+This reordering feature is available in the Content Manager and the API.
+
+## Code location
+
+`packages/core/database/lib/entity-manager/relations-orderer.js`
+
+## How order is stored in DB?
+
+- We store the order of the relations in a `order` field.
+- For bidirectional relations, we store the order of the other side in a `inverse_order` field.
+
+We store order for all type of relations, except for:
+
+- Polymorphic relations (too complicated to implement).
+- One to one relations. (as there is only one relation per pair)
+
+### Many to many (Addresses <-> Categories)
+
+<img src="/img/database/m2m-example.png" alt="many to many relation" />
+
+- `category_order` is the order of the categories relations in an address entity.
+- `address_order` is the order of the addresses relations in a category entity.
+
+### One to one (Kitchensinks <-> Tags)
+
+<img src="/img/database/o2o-example.png" alt="one to one relation" />
+
+- there is no `order` fields as there is only one relation per pair.
+
+### One way relation (Restaurants <-> Categories)
+
+Where a restaurant has many categories:
+
+<img src="/img/database/mw-example.png" alt="many way relation" />
+
+- `category_order` is the order of the categories relations in a restaurant entity.
+- There is no `restaurant_order` as it is a one way relation.
+
+## How to use this in the DB layer?
+
+See more on [Strapi Docs](https://docs.strapi.io/dev-docs/api/rest/relations#connect)
+
+The database layer receives a payload like this:
+
+```js
+  category: {
+    connect: [
+      { id: 6, position: { after: 1} },    // It should be after relation id=1
+      { id: 8, position: { end: true }},   // It should be at the end
+    ],
+    disconnect: [
+      { id: 4 }
+    ]
+  }
+```
+
+## How does it work?
+
+We use fractional indexing. This means that we use decimal numbers to order the relations.
+
+### Simple example
+
+<img src="/img/database/reordering-algo-1.png" alt="An example of reordering in the CM"/>
+
+### Complex example
+
+<img src="/img/database/reordering-algo-2.png" alt="An example of reordering in the CM" />
+
+### Algorithm steps
+
+From the `connect` array:
+
+1. For every element, **load relations by id**, **from fields `after` or `before`**.
+2. Start computing, from after before relations array:
+   1. **Initialize** with after/before relations (**step 1**). Let's call these ones **init relations.**
+   2. **Apply the updates** from the `connect` array, **sequentially**.
+      1. `before`:
+         1. Place **before** the element with that **id** in the list
+            1. If placing before an init relation: **order = beforeRelation.order -0.5**
+               _(0.5 is based on the “gap” between the init relation and the one before, so it is placed in between them)_
```

Comment:
Does this mean that the value might not always be 0.5? But is computed as the gap divided by 2?

If so maybe we should make that clearer

Reply:
It should always be 0.5 in this step. 

Placing a new relation between
{id: 1, order: 1}
{id: 2, order: 2}

Will result in
{id: 1, order: 1}
{id: 3, order: 1.5}
{id: 2, order: 2}

Saying that, I have rephrased the algo steps hoping it's a little bit clearer. wdyt?